### DISCUSSION TITLE
Suggesting some minor changes.

### DISCUSSION THREAD

**Discussion by robkam** [saveweb/wikiteam3#17] [Ideas]
- Usage should mention how to batch download wikis (Wikiteam had launcher.py with listfile).
- Also, include the wiki logo in the download folder. Including the logo in the IA upload puts the logo on the archive instead of the generic IA one.
- Instead of the user having to manually grep, when Wikiteam3dumpgenerator completes, it should check the dumped XML file and also check for a closing `</mediawiki>` tag.
- Fix the corrupt folder name, e.g. `https://subdomain.miraheze.org/w/api.php` gets a _w- in the folder name.

**Comment by yzqzss** [comment]
> batch download

Adding a launcher is unnecessary for something that can be done via simple shell script.

> wiki logo

OK. #18

> xml check

Good suggestion, but parsing xmldump is not easy, especially since it comes in various formats. If anyone is willing to PR the big feature I'll take it.

> corrupt folder name

I've **repeated many times that this is not a bug**. if you don't keep it consistent with the HTTP PATH, the identifier will not be consistent.

What do you think would happen if example.com, example.com/a/ and example.com/w/ were three different wikis?

**Reply by robkam** [reply]
> > xml check
> 
> Good suggestion, but parsing xmldump is not easy, especially since it comes in various formats. If anyone is willing to PR the big feature I'll take it.

Not a PR but something like this in my fork at https://github.com/robkam/mediawiki-dump-generator/blob/messy5/wikiteam3/utils/checkxml.py

**Reply by yzqzss** [reply]
check my reviews: https://github.com/mediawiki-client-tools/mediawiki-dump-generator/pull/221

**Reply by yzqzss** [reply]
To be honest, your code quality is not good and using regex to parse xml is not best practice, I will not merge this.

**Reply by yzqzss** [reply]
please use sax to stream parse xml.

https://github.com/saveweb/wikiteam3/blob/v4-main/tools/xml2titles.py

**Reply by robkam** [reply]
> To be honest, your code quality is not good

That's no surprise

Another suggestion, to use SevenZipFile from py7zr instead of relying on 7z in PATH?

**Reply by yzqzss** [reply]
https://github.com/miurahr/py7zr#performance

**Reply by robkam** [reply]
> please use sax to stream parse xml.

https://github.com/robkam/checkxml/blob/main/check_xml_inegrity.py

**Reply by robkam** [reply]
> > wiki logo
> 
> OK. #18

Unnecesary - wikiteam3uploader fetches and uploads the logo.

**Comment by robkam** [comment]
Shouldn't --bypass-cdn-image-compression be enabled by default since the user probably wants to preserve the original image?

**Reply by yzqzss** [reply]
?

**Reply by robkam** [reply]
Shouldn't have to specify --bypass-cdn-image-compression should just fetch the uncompressed image(s) by default.

**Reply by robkam** [reply]
Never mind I've misunderstood.
